packages/medusa/src/api-v2/admin/price-lists/[id]/products/route.ts

```diff
@@ -0,0 +1,38 @@
+import { MedusaError } from "@medusajs/utils"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../types/routing"
+import { GET as AdminGetProductsRoute } from "../../../products/route"
+import { listPriceLists } from "../../utils"
+
+export const GET = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse
+) => {
+  const id = req.params.id
+  const [[priceList], count] = await listPriceLists({
+    container: req.scope,
+    fields: ["prices.variant_id"],
+    variables: { filters: { id }, skip: 0, take: 1 },
+  })
+
+  if (count === 0) {
+    throw new MedusaError(
+      MedusaError.Types.NOT_FOUND,
+      `Price list with id: ${id} was not found`
+    )
+  }
+
+  const variantIds: string[] =
+    priceList.prices?.map((price) => price.variant_id).filter(Boolean) || []
+
+  req.filterableFields = {
+    ...req.filterableFields,
+    variants: {
+      id: variantIds,
+    },
+  }
+
+  AdminGetProductsRoute(req, res)
```

Comment:
thought: I am a bit unsure of this approach, if any changes are applied to the get product route do we really want them to be applied here too? I think a simple duplication would be better in that scenario

Reply:
Worst case this become a middleware and we apply the product route on the end point then 😋

Reply:
Yes, This one is basically a product endpoint 😄 this should really be a filter on the product endpoint but we also have this endpoint that does the same thing. 

If you feel strongly about it, I can extract the product remote query to a util and reuse it here. Wdyt?

Reply:
And this somewhat confirms my comment on the route, if we need to do something like this it means there could be an issue with the idea of fetching products for a price list the way we are

Reply:
In total agreement on that. Shall we remove this API route and stick to the filter on products then? We can make a note on the breaking changes that this route will no longer be supported on v2

Reply:
@riqwan yeah I think that would be much more intuitive, and as long as we document the breaking change it shouldn't be a big deal.